builtin/builtin.go

```diff
@@ -772,6 +772,93 @@ func builtin_len(self, v py.Object) (py.Object, error) {
 	return py.Len(v)
 }
 
+const max_doc = `
+max(iterable, *[, default=obj, key=func]) -> value
+max(arg1, arg2, *args, *[, key=func]) -> value
+
+With a single iterable argument, return its biggest item. The
+default keyword-only argument specifies an object to return if
+the provided iterable is empty.
+With two or more arguments, return the largest argument.`
+
+func builtin_max(self py.Object, args py.Tuple, kwargs py.StringDict) (py.Object, error) {
+	return min_max(args, kwargs, "max")
+}
+
+const min_doc = `
+min(iterable, *[, default=obj, key=func]) -> value
+min(arg1, arg2, *args, *[, key=func]) -> value
+
+With a single iterable argument, return its smallest item. The
+default keyword-only argument specifies an object to return if
+the provided iterable is empty.
+With two or more arguments, return the smallest argument.`
+
+func builtin_min(self py.Object, args py.Tuple, kwargs py.StringDict) (py.Object, error) {
+	return min_max(args, kwargs, "min")
+}
+
+func min_max(args py.Tuple, kwargs py.StringDict, name string) (py.Object, error) {
+	kwlist := []string{"key", "default"}
+	positional := len(args)
+	var format string
+	var values py.Object
+	var cmp func(a py.Object, b py.Object) (py.Object, error)
+	if name == "min" {
+		format = "|$OO:min"
+		cmp = py.Le
+	} else if name == "max" {
+		format = "|$OO:max"
+		cmp = py.Ge
+	}
+	var defaultValue py.Object
+	var keyFunc py.Object
+	var maxval py.Object
+
+	if positional > 1 {
+		values = args
+	} else {
+		err := py.UnpackTuple(args, nil, name, 1, 1, &values)
+		if err != nil {
+			return nil, err
+		}
+	}
+	err := py.ParseTupleAndKeywords(nil, kwargs, format, kwlist, &keyFunc, &defaultValue)
```

Comment:
I note that you haven't implemented the `default` or `key` arguments...

The default argument would be very easy to implement, just copy the `defaultValue` to `maxval` if it is not None.

The `key` argument is a little trickier...